conda/recipes/libcudf/meta.yaml

```diff
@@ -60,7 +59,7 @@ outputs:
       build:
         - cmake {{ cmake_version }}
       run:
-        - cudatoolkit {{ cuda_spec }}
+        - {{ pin_compatible('cudatoolkit', max_pin='x', min_pin='x') }}
```

Comment:
@ajschmidt8 I see you changed this to use `cuda_spec` in #10326 but I don't fully understand why `cuda_spec` was preferred over using `pin_compatible`. Any advice on this?

Reply:
Discussed offline with @ajschmidt8. The `cuda_spec` approach where we manually specify the bounds is needed in the C++ recipes like `libcudf/meta.yaml` because there is no `host:` section that `pin_compatible` can use to determine the runtime dependency bounds. That's because we have a split recipe with a top-level `requirements:` section that applies to all the recipe outputs (to build exactly once and avoid building the C++ twice for `libcudf` and `libcudf-tests`, for example).

There are several recipes in RAPIDS that are using `cuda_spec` or similar constructions where the `pin_compatible` _can_ (and should) be used instead, but this is not one of them.